### PR TITLE
Rename the ProcFS object for style, and set up panic handlers.

### PR DESCRIPTION
**Pull request by brendandburns**
Ref https://github.com/kubernetes/kubernetes/issues/12818

### CODE REVIEW

**Comment by brendandburns** [comment]
ugh, forgot to prepend "/proc" hang on...

**Comment by brendandburns** [comment]
ok, ready for review.

**Comment by brendandburns** [comment]
@dchen1107 @derekwaynecarr fyi

**Comment by derekwaynecarr** [comment]
On the Kubelet, we had been setting other sysctl flags in container manager. Now that the scope of flags has grown, it would be good if it wasn't scattered.

If the Kubelet is booting the API server from pod manifest, I would prefer the API server to not set kernel flags or values.

File: pkg/kubelet/kubelet.go [hunk 1]
```diff
@@ -319,7 +319,10 @@ func NewMainKubelet(
 		return nil, err
 	}
 
-	procFs := procfs.NewProcFs()
+	procFs := procfs.NewProcFS()
+	if err := procfs.SetupRestartOnPanicAndOops(procFs); err != nil {
```
Comment (vishh) [inline pkg/kubelet/kubelet.go]: This should be moved to within [container manager](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/cm/container_manager_linux.go#L146).

Once that is done you will no longer have to inject a procFS interface.

**Comment by brendandburns** [comment]
@vishh @derekwaynecarr thanks for the feedback, refactored this PR in the way you suggest.

ptal, thanks!
--brendan

**Comment by derekwaynecarr** [comment]
This looks good to me!

File: pkg/util/sysctl/sysctl.go [hunk 2]
```diff
@@ -27,9 +27,14 @@ const (
...
 	VmOvercommitMemory = "vm/overcommit_memory"
 	VmPanicOnOOM       = "vm/panic_on_oom"
+	KernelPanic        = "kernel/panic"
+	KernelPanicOnOops  = "kernel/panic_on_oops"
 
 	VmOvercommitMemoryAlways    = 1 // kernel performs no memory over-commit handling
 	VmPanicOnOOMInvokeOOMKiller = 0 // kernel calls the oom_killer function when OOM occurs
+
+	KernelPanicOnOopsAlways  = 1  // kernel panics on kernel oops
+	KernelPanicRebootTimeout = 10 // seconds after a panic for the kernel to reboot
```
Comment (derekwaynecarr) [inline pkg/util/sysctl/sysctl.go]: It's possible our operation team may want to tweak this in future, but this is fine for this Pr

Comment (vishh) [inline pkg/util/sysctl/sysctl.go]: We need some sort of an unified config representation for all these settings.

**Comment by vishh** [comment]
LGTM

**Comment by piosz** [comment]
@k8s-bot e2e test this please

**Pull request merged**